src/main/java/net/apnic/rdap/filter/filters/RDAPPreFilter.java

```diff
@@ -40,6 +42,8 @@ public Object run()
         RDAPRequestPath path = RDAPRequestPath.createRequestPath(
                 context.getRequest().getRequestURI());
         context.put(RequestContextKeys.RDAP_REQUEST_PATH, path);
+        context.getRequest().setAttribute(ORIGIN_ATT_KEY,
+            context.getRequest().getRequestURL().toString());
```

Comment:
Yeah working around an issue with the way spring is setting up Zuul. It wasn't my first choice doing it this way.

Basically when an error is thrown it gets redirect to the spring error path and you loose all context of the original request. You can try and piece it back together from the error path but there are no certainties made as to correctness which would be worse.